### ISSUE TITLE
slacko64 6.9.9.9 missing eudev?

### ISSUE THREAD

**Issue by ghost** [puppylinux-woof-CE/woof-CE#1717]
I see it's looking for eudev-3.2.5-1:

![Screenshot](https://user-images.githubusercontent.com/5408521/70725764-f81f6500-1cca-11ea-9c90-07881f5b9b97.png)

But I found eudev-3.2.8-2 here instead:
http://ftp.nluug.nl/ftp/pub/os/Linux/distr/puppylinux/pet_packages-common64/

Should 3.2.8-2 be used in slacko64 6.9.9.9 then? Here is the pet.specs for 3.2.8-2:
```
eudev-3.2.8-x86_64_common64-2|eudev|3.2.8-x86_64_common64-2||BuildingBlock|7520||eudev-3.2.8-x86_64_common64-2.pet||programs for dynamic creation of device nodes.|slackware64|14.0||
eudev_DEV-3.2.8-x86_64_common64-2|eudev_DEV|3.2.8-x86_64_common64-2||BuildingBlock|64||eudev_DEV-3.2.8-x86_64_common64-2.pet|+eudev|eudev development|slackware64|14.0||
```

**Comment by ghost** [comment]
I tried to make a slacko 6.9.9.9 64-bit build with eudev-3.2.8-2 and it seemed a bit slower to show the desktop and sound (retrovol) volume slider didn't work any more (always mute), but those might have been caused by something else. Will try to play around more later.

**Comment by 01micko** [comment]
I always used the slackware version. I think if upstream supports eudev then why not?

`yes|eudev|eudev|exe,dev,doc,nls`

**Comment by ghost** [comment]
Thanks 01micko! It is working well with the slackware version in my test. I'll send a pull request for it then. :+1:

**Comment by wdlkmpx** [comment]
This is the kind of feedback that really matters, building ISOs of what is available and suggesting changes to default packages. Nothing else matters.

The first option is a PR to change eudev to the slackware64 version, which is a bit old by now.

So here is an answer to the question about the missing eudev pkg.

`eudev-3.2.5-x86_64_common64-1.pet` is missing
`udev-3.2.5-x86_64_common64-1.pet` is a corrupted/incomplete pkg

All the 64-bit builds that use the common64 eudev/udev are broken. It only took a few weeks for people to realize that.

Both .pets are the same package with different names to provide the same framework for all common packages that depend on eudev. That way there's no need to compile certain packages for every single supported distro version. Although it's possible to do that, but it requires someone to do it.. and more importantly, someone to upload the pkgs.

It's also becoming apparent that slackware 14.2 (+alt xorg 1.11), dpup stretch (+alt xorg 1.11) and upup xenial should be the minimum supported versions, because time really flies and big changes in hardware and paradigms are already happening. I see people everywhere selling [extremely] cheap and old powerful hardware. Not to mention that most people nowadays prefer mobile devices.. things change.

The 64-bit eudev .pet worked fine in my tests. But I only tested the 32bit version with a pkg downloaded from ibiblio.

I remember testing the 64bit eudev/udev only with pkgs from local-repositories.. working packages. There last revision did work fine for me.

The only problem I saw with eudev 3.2.8 is that pup-volume-monitor was unable to detect changes [some unknown conflict happening].. and that is the single most important app for me... p-v-m probably needs a rewrite. Everything else seemed to work fine, but there are more udev warnings/errors and stuff that don't really seem to cause trouble.

`woof-code/rootfs-packages/backend_modprobe` seems to cause some errors/warnings with recent eudev versions in [older] distros [or kernels], it needs some proper testing... It seems that most puppy packages can be deleted in order to make the next generation improvements. I deleted and replaced lots of stuff in the whole codebase, but it's just the beginning.

These are repackaged .pets
[eudev-3.2.5-x86_64_common64-2.pet_fake.zip](https://github.com/puppylinux-woof-CE/woof-CE/files/3957824/eudev-3.2.5-x86_64_common64-2.pet_fake.zip)
[udev-3.2.5-x86_64_common64-2.pet_fake.zip](https://github.com/puppylinux-woof-CE/woof-CE/files/3957828/udev-3.2.5-x86_64_common64-2.pet_fake.zip)


Entries for common64
```
eudev-3.2.5-x86_64_common64-2|eudev|3.2.5-x86_64_common64-2||BuildingBlock|7208||eudev-3.2.5-x86_64_common64-2.pet||No description provided||||
udev-3.2.5-x86_64_common64-2|udev|3.2.5-x86_64_common64-2||BuildingBlock|7208||udev-3.2.5-x86_64_common64-2.pet||No description provided||||
```

I did not perform tests.

**Comment by wdlkmpx** [comment]
I built 3 ISOs from scratch using [e]udev-3.2.5...pet from local-repositories (the ibiblio pkgs are incomplete or missing - if this is not fixed, all yes|udev|| should be changed to yes|udev|udev):

(1) slacko64 14.2 - broken [I think changes in hardware are detected but sound and network are dead]
(2) slacko64 14.1 -  broken, changes in hardware are detected but sound and network are dead - this also needs some work with some pkgs
(3) slacko64 14.0 - ok - this is where it was compiled

(1) can be fixed by merging PR #1719 
(2) BROKEN - need to compile eudev64 3.2.5 - although I don't know if installing kmod64 from common64 makes any difference

I'll build a dpup buster64, upup tahr64 / xenial64 / bionic64 for the lolz.

I think the situation is better in 32bit builds, since the common32 eudev also works fine in dpup stretch and maybe in slacko32 14.2.. can't remember. Have to build another 5 isos to see.

**Comment by 01micko** [comment]
If just one shared lib misses one symbol chaos can happen. That's why I tend to use the native packages as much as possible, even if it isn't the 'latest and greatest'. Then I try to compile against those native libs.

While all the 'common' stuff may seem convenient it can be the cause of much headache. That's why I had separate branches of `petbuilds` that later got deleted. The more complex the program the bigger the headache.

**Comment by woodenshoe-wi** [comment]
I think the original idea I had for the “common” pets has been stretched beyond what can work.

I used **cut down** package lists to make a common32 and common64 iso and devx that only contained basic libraries that I hoped would be compatible with newer libraries. https://github.com/woodenshoe-wi/common64/blob/master/DISTRO_PKGS_SPECS-slackware64-14.0

Then I would use run_woof to compile the pet using the common32/64 iso/devx.  I would **NOT** compile pets in a regular slacko 14.0 puppy because it could pull in dependencies that might not be compatible with newer libraries.

When I looked through the logs of the pets I had built this way, I didn’t find eudev even though it was in the generic branch at the time.

The reason is that eudev would not build using the run_woof method.  Unless something has changed, it was probably compiled in a regular puppy with a full set of old libraries/kernel headers.

When using run_woof to compile pets, if the logs don’t show errors, you _probably_ can use the pet in any compatible pup.  But if you are just compiling it in an older version of puppy, all bets are off.

> That's why I had separate branches of petbuilds that later got deleted.

Use one generic petbuild, but compile it in each distro.  Not all generic petbuilds can be compiled as “common” pets, but all “common” pets use generic petbuilds.

**Comment by s243a** [comment]
> Both .pets are the same package with different names to provide the same framework for all common packages that depend on eudev. That way there's no need to compile certain packages for every single supported distro version. Although it's possible to do that, but it requires someone to do it.. and more importantly, someone to upload the pkgs.

Prior to updating a particular package one could copy the old common version to a repo with a name something like:

Puppy-common64_7

which would mean that the package should work for all 7 series pups and earlier. This way woof-CE would have better legacy support.

**Comment by wdlkmpx** [comment]
Removing tons of files and simplifying da codebase had a side effect: systemd-udev started to work. So I'm not mistaken when I say that many puppy scripts can just disappear and some magic will happen automatically.

I built many test ISOs from scratch for all the deb-based 64bit pups
Tests included the common64 eudev 3.2.5-2 (local-repositories), package and pkg db entries available in a previous comment, it should also be in a ftp where I uploaded it with -1 revision..... and also the compat distro udev

(1) tahrpup64 - totally broken - X doesn't start, the PKGS_SPECS needs fixing, specially xserver_xorg [too many errors in log]
(2) xenialpup64  - broken only with common64 eudev. Should use the common64 sudo, xterm does not run, misc issues, etc..
(3) bionicpup64  - ok
(4) busterpup64 - broken only with common64 eudev [many .pets are missing] - also needs native kernel.. this needs lots of work

(1) BROKEN - tahrpup64    - fixed by PR #1721 (systemd-udev)
(2) BROKEN - xenialpup64  - fixed by PR #1721 (systemd-udev)
(3) OK       bionicpup64  - but PR #1721 makes it use systemd-udev
(4) BROKEN - busterpup64 - fixed by PR #1721 (systemd-udev)

- tahrpup64 comes with udev 204 - probably the minimum supported version, but some common32/64 packages that depend on eudev probably require a higher version... testing is the key

- slacko64 14 & 14.1 come with udev 181.. too old

But if the PRs are merged (and the .pet uploaded).. only slacko64 14.1 will remain broken.

Some thoughts about the eudev pet pkg:

Most packages compiled in old distros -that need only a few dependencies- work ok, but I always said that udev is one of the few packages that are problematic even when everything seems to be ok. But I decided to experiment with the old distros bringing them up to date in a hacky and nasty way [it was certainly a mistake]. What you see on the forums, I have always done that, but with more extreme changes.

I found that some sort of error happens, it still runs and detects hardware changes, but some functionality is lost.. I can manually modprobe the network kdriver and run SNS, it probably needs a better recipe. But when a needed symbol is missing apps just don't run.

By updating stuff a bit, it's easy to create a framework where common apps (including the latest firefox and palemoon and probably some version of libreoffice) run. That's what I do, use the latest official firefox and palemoon [and other apps] in all the new builds, no need to compile those big apps.. it's possible to compile stuff for each distro version, but who is going to upload the packages to the repos?

It's quite easy to fix stuff and move the whole project forward -including core scripts- when all the compat distro versions include the minimum supported version for some key packages, but unfortunately it's a task that can only be achieved by a dedicated team of testers and builders.. woofce does not have a team.. of any sort.

Another way to deal with this is... [delete the old distros].

I have a simple cli/yad gui alternative to ffconvert [and other popular apps], but it doesn't support old ffmpeg versions (with deprecated syntax) and it works better with a static ffmpeg from johnvansickle [I helped him fix his static i686 build]. I really don't know the current state of ffconvert, is it broken? I don't know I don't care.

Now I'll test all 32bit ISOs
